### PR TITLE
normalize model rotation to interface limits

### PR DESCRIPTION
**Pull request by bkleiner**
adds arithmetic functions required to use a vector of `math::degrees` to store model rotations normalized to [-180,180)
picks up on #74

todo:
- [x] fix the interface limits

### CODE REVIEW

File: src/math/quaternion.hpp [hunk 1]
```diff
@@ -77,21 +77,21 @@ namespace math
       double test = x * y + z * w;
       if (test > 0.499 * unit) // singularity at north pole
       {
-        retVal.y = -static_cast<float>(2.0f * std::atan2(x, w) * 180.0f / math::constants::pi);
-        retVal.x = (math::constants::pi / 2) * 180.0f / math::constants::pi;
-        retVal.z = 0;
+        retVal.y = math::degrees(-static_cast<float>(2.0f * std::atan2(x, w) * 180.0f / math::constants::pi));
```
Comment (bloerwald) [inline src/math/quaternion.hpp]: heh, loving that *180/pi before putting it into one of the auto-converting types :D

File: src/math/trig.hpp [hunk 2]
```diff
@@ -9,15 +9,53 @@
...
+    }
+
+    inline degrees operator-() const
+    {
+      return degrees (-_);
+    }
+
+    inline degrees& operator+= (const degrees &v)
+    {
+      _ = normalize_degrees(_ + v._);
```
Comment (bloerwald) [inline src/math/trig.hpp]: You can implement one operator with the other to avoid the explicit normalize:

```
return *this = *this + v;
```

**Review by bloerwald — APPROVED** [review]

**Comment by bloerwald** [comment]
Didn't merge yet in case you **want** to do those two changes. If not, just tell me and I click merge.

File: src/math/quaternion.hpp (hunk 1 above)
Comment (bkleiner) [reply src/math/quaternion.hpp]: Hehe, just slapped the cast around it and didn’t even look at the (pre-existing) code

File: src/math/trig.hpp (hunk 2 above)
Comment (bkleiner) [reply src/math/trig.hpp]: Love it will do

**Comment by bkleiner** [comment]
Both comments are incorporated.

**Pull request merged**